docs/preview/03-Features/correlation-hierarchical.md

```diff
@@ -0,0 +1,292 @@
+6. The user receives both the **transaction ID** and **operation ID** in their final response.
+
+💡 This correlation is based on the `RequestId` and `X-Transaction-ID` HTTP request/response headers, however, you can fully configure different headers in case you need to.
+💡 The `X-Transaction-ID`  can be overridden by the request, meaning: if the HTTP request already contains a `X-Transaction-ID` header, the same header+value will be used in the HTTP response.
+
+Additional [configuration](#configuration) is available to tweak this functionality.
+
+## Installation
+
+This feature requires to install our NuGet package:
+
+```shell
+PM > Install-Package Arcus.WebApi.Logging
+```
+
+## Usage
+
+To fully benefit from the Arcus' HTTP correlation functionality, both sending and receiving HTTP endpoints should be configured.
+
+### Sending side
+
+To make sure the correlation is added to the HTTP request, following additions have to be made:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+
+WebApplication builder = WebApplication.CreateBuilder();
+builder.Services.AddHttpCorrelation(options => options.Format = HttpCorrelationFormat.Hierarchical);
+builder.Services.AddHttpClient("from-service-a-to-service-b")
+                .WithHttpCorrelationTracking();
+
+WebApplication app = builder.Build();
+```
+
+Then, the [created `HttpClient`](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests) will be enhanced with HTTP correlation tracking. This means that the request will be altered to include the HTTP correlation information and the endpoint will be tracked as a [HTTP dependency telemetry](https://observability.arcus-azure.net/Features/writing-different-telemetry-types#measuring-http-dependencies).
+
+Alternatively, an existing `HttpClient` can be used to send a correlated HTTP request:
+
+```csharp
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Arcus.WebApi.Logging.Core.Correlation;
+
+var client = new HttpClient();
+
+IHttpCorrelationInfoAccessor accessor = ... // From dependency injection.
+ILogger logger = ... // From dependency injection.
+
+var request = new HttpRequestMessage(HttpMethod.Get, "https://localhost/service-b");
+await client.SendAsync(request, accessor, logger);
+```
+
+💡 The HTTP correlation tracking can also be configured, see [this section](#configuring-http-correlation-client-tracking) for more information.
+
+### Receiving side
+
+To make sure the correlation is added to the HTTP response, following additions have to be made:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+
+WebApplicationBuilder builder = WebApplication.CreateBuilder();
+builder.Services.AddHttpCorrelation(options => options.Format = HttpCorrelationFormat.Hierarchical);
+
+WebApplication app = builder.Build();
+app.UseHttpCorrelation();
+app.UseRouting();
+app.UseRequestTracking();
+```
```

Comment:
We still have to come up with a useful solution for providing the Application Insights connection string once (see https://github.com/arcus-azure/arcus.observability/issues/471) and the application name should be passed along (see https://github.com/arcus-azure/arcus.observability/issues/470) but those are Application Insights thingies, only implicitly related to the W3C changes made here.
But otherwise, yes, W3C is with this change made by default, so any Arcus-usage on built-in Microsoft stuff on the sending side could then be removed as it will be done automatically.

Reply:
Also didn't verify yet in Azure Functions HTTP trigger. Probably will have to do something to disable the request tracking module there, too.

Reply:
The connection-string related stuff is actually internal only.  I mean, the user / consumer of Arcus is not impacted ?

Reply:
Depends on how you see it, currently, you'll have to add it twice if you want automatic dependency tracking as the Observability library doesn't provide a way (yet) to do this once. So, no, impacted on existing functionality; yes, impacted if you want to include automatic dependency tracking.

Reply:
If I'm not mistaken, this is not really obvious from the documentation, or did I miss something ?

Reply:
Your comment here is on the `correlation-hierarchical.md`  file, which is the one where the old system is explained. `correlation.md` is where the W3C is explained. 
We can't yet provide a link to the Observability part as it is not implemented yet.

Reply:
Or maybe you didn't bother as you're sure you'll find a solution to it :)

Reply:
😄 , appreciate the trust.
Currently, the best I can think of is updating the `.WriteTo.AzureApplicationInsightsWithConnectionString(...)` with an additional `IServiceProvider` so that the connection string can be 'late' assigned to the registered `TelemetryClient`.